util/src/time.cc

```diff
@@ -0,0 +1,66 @@
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <leatherman/util/time.hpp>
+
+namespace leatherman { namespace util {
+
+    std::string getExpiryDatetime(int expiry_minutes) {
+        struct tm expiry_time_info;
+        char expiry_time_buffer[80];
+
+        // Get current time and add the specified minutes
+        time_t expiry_time { time(nullptr) };
+        expiry_time += 60 * expiry_minutes;
+
+        // Get local time structure
+        get_local_time(&expiry_time, &expiry_time_info);
+
+        // Return the formatted string
+        if (strftime(expiry_time_buffer, 80, "%FT%TZ", &expiry_time_info) == 0) {
+            // invalid buffer
+            return "";
+        }
+
+        return std::string { expiry_time_buffer };
+    }
+
+    void displayProgress(double percent, int len, std::string status) {
+        printf("%c[2K", 27);
```

Comment:
This adds colorization, we should make this work on Windows or note it won't be supported. @peterhuene I'm not particularly familiar with terminal colorization, how do I read this?

@Magisus you can look at the colorization handling in logging for some guidance.